### PR TITLE
chore: skip publishing some unnecessary files

### PR DESCRIPTION
**Pull request by nlf**
at a glance, these do not appear necessary to ship with the tarball and don't really make sense for most end users.

these changes reduce the published tarball by ~300kb and the installed size by ~1mb

### CODE REVIEW

**Comment by ruyadorno** [comment]
I'd like to give it a try at our release integration with nodejs to make sure it doesn't prune any of the expected files there, for instance `.eslintrc.json` looks like it's pb going to be required for tests to run there, it's fine if we want to tweak it but let's make sure it doesn't break that workflow 😊

**Comment by nlf** [comment]
the node release integration clones this repo, it doesn't use the tarball. It can't since the tarball doesn't include the tests themselves either

File: .npmignore [hunk 1]
```diff
@@ -15,8 +15,19 @@ node_modules/npm-registry-mock
 /coverage/
 
 # don't need these in the npm package.
+Makefile
+.eslintrc.json
+.licensee.json
+.mailmap
+changelogs/*
+configure
+make.bat
```
Comment (isaacs) [inline .npmignore:23]: Had to check to make sure that the `install.sh` isn't calling `./configure` or `make`, but these are fine to leave out.

**Review by isaacs — APPROVED** [review]
yes

**Pull request merged**